deps/childprocess.lua

```diff
@@ -164,4 +164,61 @@ local function spawn(command, args, options)
   return em
 end
 
+local function exec(command, args, options, callback)
+  local file
+  local stringArgs = command..table.concat(args, ' ')
+  if process.globalProcess().env._system_name:lower() == 'win32' then
+    file = 'cmd.exe'
+    args = {'/s', '/c', '"'..stringArgs..'"' }
+  else
+    file = '/bin/sh'
+    args = {'-c', stringArgs }
+  end
+
+  if options.shell then file = options.shell end
+  if not options.maxBuffer then options.maxBuffer = 200 * 1024 end
+
+  local stdout, stderr = {}, {}
+  local exitCode, err
+  local called = 2
+  local function done()
+    called = called - 1
+    if called == 0 then
+      if exitCode == 0 then
+        err = nil
+      elseif not exitCode then
+        err = 'Command failed or was killed'
+      else
+        err = exitCode
+      end
+      return callback(err, table.concat(stdout, ""), table.concat(stderr, ""))
+    end
+  end
+  local function onClose(_exitCode)
+    exitCode = _exitCode
+    done()
+  end
+
+  local child = spawn(file, args, options)
+
+  child.stdout:on('data', function(chunk)
+    if #stdout > options.maxBuffer then
```

Comment:
you will need to keep track of the number of bytes read for each chunk type. #stdout isn't going to give you the right number.